### ISSUE TITLE
[a11y]: CodeSnippet Multiline Show more obscures focus element and potentially fails WCAG 2.4.11

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19273]
### Package

@carbon/react, @carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

N/A

### Description

When Show more is activated in CodeSnippet Multiline, the focused element (Show more/less button) may move out of view depending on the page zoom and/or length of the snippet. If the element currently in focus is not at least partially visible in the user's viewport, this _may_ fail 2.4.11: Focus Not Obscured (Minimum) as part of WCAG 2.2. This occurs in both React and WC versions.

### WCAG 2.1 Violation

2.4.11: Focus Not Obscured (Minimum)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-codesnippet--multiline

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-codesnippet--multiline
2. Zoom in to ~250%
3. Press tab until the Show more button is in focus
4. Activate button (press SPACE) and observe that the element in focus is not visible in the viewport

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by warrenmblood** [comment]
Found this in the WCAG 2.2 Understanding Docs which I think exempts this issue from a violation, but would appreciate the team's thoughts

![Image](https://github.com/user-attachments/assets/3588fecd-c3d4-44f1-b293-1b1094b80cc7)

**Comment by warrenmblood** [comment]
I agree that retaining the scroll location, as the component currently does, is probably best here. Closing this issue out since we have agreement that this isn't a WCAG 2.4.11 violation under the exception I pointed out above.
